controller/tenant.go

```diff
@@ -175,34 +182,84 @@ func (c *TenantController) Update(ctx *app.UpdateTenantContext) error {
+	// restrict access to users with a `featureLevel` set to `internal`
+	user, err := c.getCurrentUser(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{"auth_url": c.authURL}, "unable get current user")
+		return config, err
+	}
 
-		tenantConfig := js.GetPath("data", "attributes", "contextInformation", "tenantConfig")
-		if tenantConfig.Interface() != nil {
-			cheVersion := getJsonStringOrBlank(tenantConfig, "cheVersion")
-			jenkinsVersion := getJsonStringOrBlank(tenantConfig, "jenkinsVersion")
-			teamVersion := getJsonStringOrBlank(tenantConfig, "teamVersion")
-			mavenRepoURL := getJsonStringOrBlank(tenantConfig, "mavenRepo")
-			return config.WithUserSettings(cheVersion, jenkinsVersion, teamVersion, mavenRepoURL), nil
+	if user.Data.Attributes.FeatureLevel != nil && *user.Data.Attributes.FeatureLevel == "internal" {
+		log.Debug(ctx,
+			map[string]interface{}{
+				"auth_url":      auth.ShowUserPath(),
+				"user_name":     *user.Data.Attributes.Username,
+				"feature_level": *user.Data.Attributes.FeatureLevel},
+			"user is allowed to update tenant config")
+		if tenantConfig, exists := user.Data.Attributes.ContextInformation["tenantConfig"]; exists {
+			if tenantConfigMap, ok := tenantConfig.(map[string]interface{}); ok {
+				var cheVersion, jenkinsVersion, teamVersion, mavenRepoURL *string
+				if v, ok := tenantConfigMap["cheVersion"].(string); ok {
+					cheVersion = &v
+				}
+				if v, ok := tenantConfigMap["jenkinsVersion"].(string); ok {
+					jenkinsVersion = &v
+				}
+				if v, ok := tenantConfigMap["teamVersion"].(string); ok {
+					teamVersion = &v
+				}
+				if v, ok := tenantConfigMap["mavenRepo"].(string); ok {
+					mavenRepoURL = &v
+				}
+				return config.WithUserSettings(cheVersion, jenkinsVersion, teamVersion, mavenRepoURL), nil
+			}
+		}
+	} else {
+		curentFeatureLevel := "undefined"
+		if user.Data.Attributes.FeatureLevel != nil {
+			curentFeatureLevel = *user.Data.Attributes.FeatureLevel
 		}
+		log.Error(ctx,
+			map[string]interface{}{
+				"auth_url":      auth.ShowUserPath(),
+				"user_name":     *user.Data.Attributes.Username,
+				"feature_level": curentFeatureLevel},
+			"user is not allowed to update tenant config")
 	}
 	return config, nil
 }
 
+func (c *TenantController) getCurrentUser(ctx context.Context) (*auth.User, error) {
+	log.Info(ctx, map[string]interface{}{"auth_url": c.authURL, "http_client_transport": reflect.TypeOf(c.httpClient.Transport)}, "retrieving user's profile...")
+	authClient, err := newAuthClient(ctx, c.httpClient, c.authURL)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{"auth_url": c.authURL}, "unable to parse auth URL")
+		return nil, err
+	}
+	resp, err := authClient.ShowUser(ctx, auth.ShowUserPath(), nil, nil)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{"auth_url": auth.ShowUserPath()}, "unable to get user info")
+		return nil, errs.Wrapf(err, "failed to GET %s due to error", auth.ShowUserPath())
+	}
+	if err != nil {
```

Comment:
... just in case an err is set since the previous check?

Reply:
yes ;) joke aside, let me remove that check!